specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-04-01-preview/TenantActivityLogAlerts.json

```diff
@@ -0,0 +1,547 @@
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AlertRuleLeafCondition"
+        }
+      ],
+      "properties": {
+        "anyOf": {
+          "title": "An Activity Log Alert rule 'anyOf' condition.",
+          "description": "An Activity Log Alert rule condition that is met when at least one of its member leaf conditions are met.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AlertRuleLeafCondition"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "AlertRuleLeafCondition": {
+      "description": "An Activity Log Alert rule condition that is met by comparing the field and value of an Activity Log event.\nThis condition must contain 'field' and either 'equals' or 'containsAny'.",
+      "type": "object",
+      "properties": {
+        "field": {
+          "description": "The name of the Activity Log event's field that this condition will examine.\nThe possible values for this field are (case-insensitive): 'resourceId', 'category', 'caller', 'level', 'operationName', 'resourceGroup', 'resourceProvider', 'status', 'subStatus', 'resourceType', or anything beginning with 'properties'.",
+          "type": "string"
+        },
+        "equals": {
+          "description": "The value of the event's field will be compared to this value (case-insensitive) to determine if the condition is met.",
+          "type": "string"
+        },
+        "containsAny": {
+          "description": "The value of the event's field will be compared to the values in this array (case-insensitive) to determine if the condition is met.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "ActionList": {
+      "type": "object",
+      "description": "A list of Activity Log Alert rule actions.",
+      "properties": {
+        "actionGroups": {
+          "description": "The list of the Action Groups.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ActionGroup"
+          },
+          "x-ms-identifiers": [
+            "actionGroupId"
+          ]
+        }
+      }
+    },
+    "ActionGroup": {
+      "type": "object",
+      "description": "A pointer to an Azure Action Group.",
+      "properties": {
+        "actionGroupId": {
+          "description": "The resource ID of the Action Group. This cannot be null or empty.",
+          "type": "string"
+        },
+        "webhookProperties": {
+          "type": "object",
+          "description": "the dictionary of custom properties to include with the post operation. These data are appended to the webhook payload.",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

ARM recommends against the use of the additionalProperties/empty/unschematized object pattern. ARM favors stronger typing wherever possible because it allows all consumers of your API to reason about it. This has benefits for everyone that interacts with your API including basic infrastructure, other Azure features, SDK owners, and customers. Unschematized objects are essentially API contracts that bypass Azure versioning infrastructure.

For example, customers will not be able to manage these objects using Azure Policy, or query them using ARG, both of which are standard Azure features. Customers expect to be able to use all common Azure features against all of their resources. 

Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

Reply:
I'm still not sure I understand what you want here. Do you mean that you want this property bag to be the same as activityLogAlerts' one?